### PR TITLE
patch for avoiding call sparql endpoint during execution of semantic …

### PR DESCRIPTION
**Pull request by gcornacchia**
…queries.

This is the pull request for what discussed  [here](http://wikimedia.7.x6.nabble.com/Performance-issues-with-external-triple-store-td5053674.html).
In our experience this patch improves significantly the performance of displaying a page that contains a lot of sparql queries.

### CODE REVIEW

File: src/SPARQLStore/SPARQLStore.php [hunk 1]
```diff
@@ -270,11 +270,15 @@ public function doSparqlDataDelete( DataItem $dataItem ) {
 	 * @since 1.6
 	 */
 	public function getQueryResult( Query $query ) {
-
+		$callEndpoint = $GLOBALS['smwgCallSparqlEndpointInQuery'];
 		$result = null;
 
 		if ( \Hooks::run( 'SMW::Store::BeforeQueryResultLookupComplete', array( $this, $query, &$result ) ) ) {
-			$result = $this->fetchQueryResult( $query );
+			if ($callEndpoint == true) {
```
Comment (JeroenDeDauw) [inline src/SPARQLStore/SPARQLStore.php]: `if ( $GLOBALS['smwgCallSparqlEndpointInQuery'] ) {`

File: src/SPARQLStore/SPARQLStore.php [hunk 2]
```diff
@@ -270,11 +270,15 @@ public function doSparqlDataDelete( DataItem $dataItem ) {
...
+				$result = $this->fetchQueryResult( $query );
+			} else {
+				$result = $this->baseStore->getQueryResult($query);
```
Comment (JeroenDeDauw) [inline src/SPARQLStore/SPARQLStore.php]: `( $query )`

**Comment by JeroenDeDauw** [comment]
Thanks for contributing!

**Comment by mwjames** [comment]
> In our experience this patch improves significantly the performance of displaying a page that contains a lot of sparql queries.

While I'm not opposed to this change, I'm __burning to see__ an example (on the sandbox which can be anonymized) to see the effect and possible ramifications.

The sandbox itself doesn't use the `SPARQLStore` but I can use the content on my local instance (which uses Sesame) to run some tests.

**Comment by JeroenDeDauw** [comment]
Sandbox: http://sandbox.semantic-mediawiki.org

**Comment by gcornacchia** [comment]
Any news?

**Comment by mwjames** [comment]
> Any news?

I'm still waiting on an example to evaluate the influence and have a reference in terms of possible improvements in future.

I'm also wondering about introducing a new setting, what about using `smwgSparqlQueryEndpoint = false` instead to make a switch.

**Comment by mwjames** [comment]
>I'm also wondering about introducing a new setting, what about using smwgSparqlQueryEndpoint = false instead to make a switch.

Superseded by #2159.

**Pull request closed without merging**